flink-runtime/src/test/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPoolTest.java

```diff
@@ -549,6 +551,157 @@ public void testConsistentAvailability() throws Exception {
+
+        testUseAllBuffersForSingleChannel(12, 10, maxOverdraftBuffers);
+        testUseAllBuffersForMultiChannel(12, 10, maxOverdraftBuffers);
+    }
+
+    private void testUseAllBuffersForSingleChannel(
+            int poolSize, int maxBuffersPerChannel, int maxOverdraftBuffers) throws Exception {
+        int numberOfSubpartitions = 2;
+        checkArgument(maxBuffersPerChannel > poolSize / numberOfSubpartitions);
+        LocalBufferPool bufferPool =
+                new LocalBufferPool(
+                        networkBufferPool,
+                        1,
+                        Integer.MAX_VALUE,
+                        numberOfSubpartitions,
+                        maxBuffersPerChannel,
+                        maxOverdraftBuffers);
+        bufferPool.setNumBuffers(poolSize);
+
+        // Request all buffers inside the buffer pool
+        AutoCloseableRegistry channelCloseableRegistry = new AutoCloseableRegistry();
+        for (int i = 0; i < maxBuffersPerChannel; i++) {
+            assertRequestedOverdraftBufferAndIsAvailable(bufferPool, 0, true);
+            BufferBuilder bufferBuilder = bufferPool.requestBufferBuilder(0);
+            assertNotNull(bufferBuilder);
+            channelCloseableRegistry.registerCloseable(bufferBuilder);
+        }
+
+        AutoCloseableRegistry poolCloseableRegistry = new AutoCloseableRegistry();
+        for (int i = maxBuffersPerChannel; i < poolSize; i++) {
+            assertRequestedOverdraftBufferAndIsAvailable(bufferPool, 0, false);
+            BufferBuilder bufferBuilder = bufferPool.requestBufferBuilder(0);
+            assertNotNull(bufferBuilder);
+            poolCloseableRegistry.registerCloseable(bufferBuilder);
+        }
+
+        // request overdraft buffer
+        AutoCloseableRegistry overdraftCloseableRegistry = new AutoCloseableRegistry();
+        assertRequestedOverdraftBufferAndIsAvailable(bufferPool, 0, false);
+        for (int i = 0; i < maxOverdraftBuffers; i++) {
+            BufferBuilder bufferBuilder = bufferPool.requestBufferBuilder(0);
+            assertNotNull(bufferBuilder);
+            overdraftCloseableRegistry.registerCloseable(bufferBuilder);
+            assertRequestedOverdraftBufferAndIsAvailable(bufferPool, i + 1, false);
+        }
+
+        BufferBuilder bufferBuilder = bufferPool.requestBufferBuilder(1);
+        assertNull(bufferBuilder);
+        assertRequestedOverdraftBufferAndIsAvailable(bufferPool, maxOverdraftBuffers, false);
+
+        // release all bufferBuilder
+        overdraftCloseableRegistry.close();
+        assertRequestedOverdraftBufferAndIsAvailable(bufferPool, 0, false);
+        poolCloseableRegistry.close();
+        assertRequestedOverdraftBufferAndIsAvailable(bufferPool, 0, false);
+        channelCloseableRegistry.close();
+        assertRequestedOverdraftBufferAndIsAvailable(bufferPool, 0, true);
+
+        bufferPool.lazyDestroy();
+    }
+
+    private void testUseAllBuffersForMultiChannel(
```

Comment:
I can be mistaken but this method looks the same as above (`testUseAllBuffersForSingleChannel`). I think you can merge them into one but where you want to use the different channels you can use (`i % numberOfChannel`)